### PR TITLE
Sidewinder Again: Now With Sprites!

### PR DESCRIPTION
**Pull request by Mqiib**
# Document the changes in your pull request

Adds a new mech called the "Sidewinder." Its stats are as follows (compared to gygax in parentheses)
Max Integrity: 200 (gygax 250)
Armor: MELEE = 40, BULLET = 20, LASER = 20, ENERGY = 15, BOMB = 40, BIO = 100, RAD = 75, FIRE = 100, ACID = 100
(gygax MELEE = 25, BULLET = 20, LASER = 30, ENERGY = 15, BOMB = 0, BIO = 100, RAD = 40, FIRE = 100, ACID = 100)
step_in/move speed: 2.5 (gygax 3, odysseus 2)
Base melee force: 15 (gygax 25)
Melee cooldown: 0.7 seconds (gygax 1.0 s)
Internal damage threshold: 25 (gygax 35) (% hp at which internal damage can occur)
Deflect chance: 10% (gygax 5%)

**Special:**
- Omni-directional attacks! Standing behind this mech is not safe, as it can quickly twist around to smack you in the back of the face!
- Turn on a dime! Clicking will immediately rotate the mech to face the click direction. This is temporarily disabled by EMPs, allowing you to pump it full of lead when you want it dead.
- No guns! This mech is unable to mount ranged weaponry, but melee weapons and other tools are still fair game. Keep your distance if you can!

- [x] Add primary mech sprites
- [x] sidewinder.dm
- [x] Get part/construction sprites so I can stop using the gygax
- [x] Background allowable weapon tweaks
- [x] Stop my local servers from imploding (again)
- [x] Internal screaming
- [x] Tech nodes n stuff
- [x] Add mech part/board designs
- [x] Make sure it works
- [x] Make sure it works _as intended_

# Spriting

Normal:
image

Open:
image

Thank you to @cuackles for these lovely sprites!
Construction/individual part sprites coming soon!

Destroyed:
image
(It twitches)

Thank you to @ynot01 for slamming out some construction/parts/destroyed sprites for me!

# Wiki Documentation

Equipment slots: 3

Movement speed: Fast

Special properties: Omni-directional attacks, can turn instantly via clicking rather than having to move, and can only mount melee weapons

Armor:  40 melee, 20 bullet, 20 laser, 15 energy, 40 bomb, 75 rad. 25.000°K max temperature.

Integrity: 200

Description: A squat, nimble, and rather confining exosuit taking advantage of advanced synthetic muscles to move around. Originally used for entertainment in mech "sports", many have found it to be well-suited for more practical applications.

Base construction cost: 115000 iron, 33000 titanium, 15000 glass 24000 plasma, 4000 gold
Requires 5 sheets of plastitanium for inner armor layer.

# Changelog

:cl:  Mqiib, Cuackles, ynot01
rscadd: Adds the new Sidewinder mech, for those mech-sword-fighting enthusiasts!
tweak: Code changes to how mechs check for allowed weapon modules, no gameplay impact
imageadd: Sprites for the new mech
spellcheck: adds "the" to a few mecha construction steps

/:cl:

### CODE REVIEW

File: code/game/mecha/mech_fabricator.dm [hunk 1]
```diff
@@ -71,6 +72,14 @@
 								"IPC Components",
 								"Misc"
 								)
+	var/list/combat_parts = list(
```
Comment (ynot01) [inline code/game/mecha/mech_fabricator.dm]: Unused variable

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -790,6 +790,15 @@
 	"phazon_peri", "phazon_targ", "phazon_armor")
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
 
+/datum/techweb_node/sidewinder
+	id = "mech_sidewinder"
+	display_name = "EXOSUIT: Sidewinder"
+	description = "Sidewinder exosuit designs"
+	prereq_ids = list("adv_mecha", "adv_weaponry", "mech_gygax")
```
Comment (ynot01) [inline code/modules/research/techweb/all_nodes.dm]: Why does it require gygax research? Also, gygax node already requires adv_mecha so it's redundant here

**Comment by riquewright** [comment]
Looks like clean fun, but no active ability? (Or am I blind?)

File: code/game/mecha/mech_fabricator.dm (hunk 1 above)
Comment (Mqiib) [reply code/game/mecha/mech_fabricator.dm]: ah yeah thats just a holdover from it being an old branch that i had to update

File: code/modules/research/techweb/all_nodes.dm (hunk 2 above)
Comment (Mqiib) [reply code/modules/research/techweb/all_nodes.dm]: It's a sort of sidegrade to the gygax, but ill get rid of the advanced mech requirement.

Comment (ynot01) [reply code/modules/research/techweb/all_nodes.dm]: I would think that a "sidegrade" would be equally as difficult to get, not strictly harder

Comment (riquewright) [reply code/modules/research/techweb/all_nodes.dm]: I like the sidegrade idea, you can't really have an alternate version if you never made the first one. But maybe make it 1.5k points instead?

Comment (Mqiib) [reply code/modules/research/techweb/all_nodes.dm]: 2500 is just the standard cost for regular nodes, I'd rather keep the mech designs at that price. I removed the advanced mecha node.

**Review by ynot01 — APPROVED** [review]
Code is fine

**Comment by Mqiib** [comment]
> Looks like clean fun, but no active ability? (Or am I blind?)

It's special thing is the passive ability to turn on click. It does not get any active ability.

**Review by Ebin-Halcyon — APPROVED** [review]
Incredible, absolutely phenomenal.

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**